sunpy/net/base_client.py

```diff
@@ -445,8 +445,7 @@ def search(self, *args, **kwargs):
         """
 
     @abstractmethod
```

Comment:
Was this not in another PR?

Reply:
Errr, the other PR made the changes required to make this work, this is an extension of it. This changes the base_client to describe the API you need to write a Fido client.